### PR TITLE
Fix backup folder naming and freshness edge cases; strengthen the admin debug tests

### PR DESCRIPTION
**Pull request by stefan-burke**
Knocks out two small TODO sections, each deleted from `TODO.md` in full.

## Backup storage edge cases (from the PR #1837 review)

**Every database now gets a proper, distinct backup folder.** `dbName` used to return an empty name for a parseable local `file:` URL (filing backups under a bare `/`), and it stripped the first dashed part from *every* hostname — so two different databases at `alpha-db.example.com` and `beta-db.example.com` would both be filed under `db`. It now reads the host kind from the shared `databaseHostFor` helper: local and unparseable URLs are filed under `local`, only Bunny DB hostnames have their UUID prefix dropped, and every other host keeps its full first hostname segment. Bunny and Turso names come out exactly as before.

**A backup dated in the future no longer counts as fresh forever.** The update gate blocks upgrades unless a backup was taken within the last hour, but `hasRecentBackup` only checked `now - takenAt < maxAge` — a wrongly future-dated file has a negative age, so it passed that check permanently. The freshness rule now lives in a pure `backupIsFresh` function that also rejects any timestamp more than five minutes ahead of the clock. The five-minute allowance keeps a just-taken backup valid when it was made on another machine whose clock runs slightly ahead; the allowance itself stays private to the module, and the tests restate it as their own literal so changing it is a conscious test edit.

Regression tests for all three broken cases were written first and confirmed failing before the fix, plus deterministic boundary tests on the pure `backupIsFresh`. Mutation score on the module: 58/58 killed.

## Admin debug test follow-ups (from the PR #1875 review)

Both are test-only assertion strengthenings:

- The admin Sentry test now records the envelope the page actually sends and checks the test message and its `source`/`test` tags, instead of only counting that one request happened.
- The debug page structure test now asserts the exact ordered list of section headings — an independent literal list, so a dropped, renamed, or added section is a conscious test edit — instead of counting presentation CSS classes. Writing that list immediately surfaced the Email section, which the old class counts said nothing about.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_012ehWXhDVAmSZFkKWVo4grA

### CODE REVIEW

File: src/shared/db/backup-storage.ts [hunk 1]
```diff
@@ -61,6 +59,26 @@ export const backupTimestamp = (date = new Date()): string =>
...
+ * fresh. Out-of-band backups are taken on other machines whose clocks can run
+ * a little ahead of ours; anything further ahead than this is not fresh, so a
+ * wrongly future-dated file cannot satisfy the gate forever. Five minutes.
+ */
+const BACKUP_CLOCK_SKEW_MS = 5 * 60 * 1000;
+
+/** True when a backup taken at `takenAtMs` satisfies the freshness gate at
+ *  `nowMs`: younger than `maxAgeMs`, and no further ahead of the clock than
+ *  the skew allowance. */
+export const backupIsFresh = (
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply src/shared/db/backup-storage.ts:73]: Agreed and done in follow-up PR #2032 (this PR merged before the fix could ride along). `backupIsFresh` is module-private again, and the boundary tests now drive `hasRecentBackup` itself with a pinned `FakeTime` clock and real seeded backup files — same exact window/skew edges, proved through the production read path. Mutation score on the module stays 58/58.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/shared/db/backup-storage.ts [hunk 2]
```diff
@@ -12,23 +12,21 @@ export const isRemoteDatabase = (): boolean =>
...
+  const host = databaseHostFor(url);
+  if (host === "local") return "local";
 
-  const host = new URL(url).hostname;
-  const first = host.split(".")[0]!;
-
-  // Turso hostnames: {db-name}-{org}.turso.io — the full first segment is unique
-  if (host.endsWith(".turso.io")) return first;
+  const first = new URL(url).hostname.split(".")[0]!;
+  if (host !== "bunny") return first;
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply src/shared/db/backup-storage.ts:25]: Agreed and done in follow-up PR #2032. `dbName` now strips the first dashed part only when the hostname ends in `.lite.bunnydb.net`; any other Bunny-owned hostname keeps its full first segment like every other remote host. Regression test: `dbName("libsql://alpha-db.bunnydb.net")` → `"alpha-db"`.

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (stefan-burke) [reply src/shared/db/backup-storage.ts:25]: Not actionable for any real deployment, so no migration is planned. Every production database is either a `{uuid}-{name}.lite.bunnydb.net` or `{db}-{org}.turso.io` host, and both of those namespaces are byte-identical before and after this change (the existing tests pin them). The relocated cases are a local `file:` URL — a dev-only shape whose old namespace was a bare `/`, arguably never a working folder — and self-hosted dashed hostnames, which the fleet does not contain. Even where relocation did apply, the old files stay in storage untouched (only listing/pruning/gating look at the computed folder), and the hourly freshness gate is satisfied by taking one new backup. If a non-Bunny/non-Turso host ever becomes a supported deployment shape, its naming gets decided then.

---
_Generated by [Claude Code](https://claude.ai/code)_
